spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,158 @@
+        expect(response).to render_template :new
+      end
+    end
+  end
+
+  describe 'action #update' do
+    context 'valid attributes' do
+      it 'located the requested user' do
+        put :update, params: { id: @user.id,
+                               user: { first_name: @user.first_name,
+                                       last_name: @user.last_name,
+                                       email: @user.email } }
+        expect(assigns(:user)).to eq(@user)
+      end
+
+      it "changes user's attributes" do
+        put :update, params: { id: @user.id,
+                               user: { first_name: 'Firstname',
+                                       last_name: 'Lastname',
+                                       email: 'email@mail.com' } }
+        @user.reload
+        expect(@user.first_name).to eq('Firstname')
+        expect(@user.last_name).to eq('Lastname')
+        expect(@user.email).to eq('email@mail.com')
+      end
+
+      it 'redirects to the updated user' do
+        put :update, params: { id: @user.id,
+                               user: { first_name: 'Firstname',
+                                       last_name: 'Lastname',
+                                       email: 'email@mail.com' } }
+        expect(response).to have_http_status(302)
+        expect(response).to redirect_to account_users_path
+      end
+    end
+
+    context 'invalid attributes' do
+      it 'located the requested user' do
+        put :update, params: { id: @user.id,
+                               user: { first_name: @user.first_name,
+                                       last_name: @user.last_name,
+                                       email: @user.email } }
+        expect(assigns(:user)).to eq(@user)
+      end
+
+      it "does not change user's attributes" do
+        put :update, params: { id: @user.id,
+                               user: { first_name: nil,
+                                       last_name: nil,
+                                       email: nil } }
+        @user.reload
+        expect(@user.first_name).to_not eq(nil)
+        expect(@user.last_name).to_not eq(nil)
+        expect(@user.email).to_not eq(nil)
+      end
+
+      it 're-renders the edit method' do
+        put :update, params: { id: @user.id,
+                               user: { first_name: nil,
+                                       last_name: nil,
+                                       email: nil } }
+        expect(response).to have_http_status(200)
+        expect(response).to render_template :edit
+      end
+    end
+  end
+
+  describe 'action destroy' do
+    it 'deletes the user' do
+      expect { delete :destroy, params: { id: @user.id } }.to change(User, :count).by(-1)
+    end
+
+    it 'redirects to user#index' do
+      delete :destroy, params: { id: @user.id }
+      expect(response).to have_http_status(302)
+      expect(response).to redirect_to account_users_path
```

Comment:
there's also way to test flash messages, e.g.:
```
expect(controller).to set_flash
expect(controller).to set_flash[:success]
expect(controller).to set_flash[:alert].to(/are not valid/).now
```

Reply:
it's ok to try, but not required for all places.
personally I'm not nesting flash messages in my projects